File: android/app/src/main/java/com/psy/data/remote/BackupApi.kt
package com.psy.data.remote

import com.psy.data.remote.dto.BackupRequest
import com.psy.data.remote.dto.BackupResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface BackupApi {
    @POST("backup")
    suspend fun upload(@Body req: BackupRequest)

    @GET("backup")
    suspend fun download(): Response<BackupResponse>
}

File: ios/Psy/Data/Remote/BackupAPI.swift
import Foundation
struct BackupAPI {
    let client: APIClient
    func upload(blob: String) async throws {
        try await client.postNoContent("backup", body: BackupRequest(blob: blob), authed: true)
    }
    func download() async throws -> BackupResponse? {
        try await client.get("backup", authed: true)
    }
}
